src/main/java/org/openstreetmap/atlas/checks/event/CheckFlagEvent.java

```diff
@@ -52,27 +55,29 @@ public static JsonObject flagToFeature(final CheckFlag flag,
         additionalProperties.forEach(flagProperties::addProperty);
 
         final JsonObject feature;
-        final List<LocationIterableProperties> geometries = flag.getLocationIterableProperties();
+        final List<GeoJsonBuilder.GeometryWithProperties> geometries = flag
+                .getLocationIterableProperties();
         if (geometries.size() == 1)
         {
             feature = GEOJSON_BUILDER.create(geometries.get(0));
         }
         else
         {
-            feature = GEOJSON_BUILDER.createGeometryCollection(geometries).jsonObject();
+            feature = GEOJSON_BUILDER.createGeometryCollectionFeature(geometries).jsonObject();
         }
-
         final JsonArray featureProperties = new JsonArray();
         final Set<JsonElement> featureOsmIds = new HashSet<>();
-        geometries.stream().forEach(
-                geometry -> Optional.ofNullable(geometry.getProperties()).ifPresent(propertyMap ->
-                {
-                    final JsonObject properties = new JsonObject();
-                    propertyMap.forEach(properties::addProperty);
-                    featureProperties.add(properties);
-
-                    Optional.ofNullable(properties.get("osmid")).ifPresent(featureOsmIds::add);
-                }));
+        // If flagged object is relation then populate relation member properties
+        if (!flag.getFlaggedRelations().isEmpty())
+        {
+            populateFlaggedRelationProperties(flag, flagProperties, geometries, featureOsmIds,
+                    featureProperties);
+        }
+        else
+        {
+            populateFlaggedObjectProperties(geometries, featureOsmIds, featureProperties);
```

Comment:
It looks like non-relations will only be added to the geojson if there are no relations in the flagged objects. We want to always add all objects.

Reply:
This is a mistake from my part. Initially I populated geometries separately. But in this version, the geometries are populated from both FlaggedObjects and FlaggedRelations [this method](https://github.com/sayas01/atlas-checks/blob/ca038e97fc53ea636beae658224d0a6caa95664a/src/main/java/org/openstreetmap/atlas/checks/flag/CheckFlag.java#L333). I will correct this.

Reply:
All object and relation properties if any,  are now populated.